fplc_flow_diagram.py

```
@@ -0,0 +1,278 @@
+        lines = text.split('\n')
+        y_offset = -(len(lines) - 1) * 12
+        
+        for line in lines:
+            text_item = self.scene.addText(line, QFont("Arial", 12, QFont.Weight.Bold))
+            text_item.setDefaultTextColor(Qt.GlobalColor.white)
+            text_width = text_item.boundingRect().width()
+            text_height = text_item.boundingRect().height()
+            text_item.setPos(x - text_width/2, y + y_offset - text_height/2)
+            y_offset += 24
+
+    def draw_flow_line(self, start_pos, end_pos, color=QColor('#2c3e50')):
```

Ruff (0.12.2)

Do not perform function call `QColor` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)